### DISCUSSION TITLE
Clicking "Open SQL Script" opens a new script window instead of file select dialog

### DISCUSSION THREAD

**Discussion by JTMosaic** [dbeaver/dbeaver#34286] [Q&A]
### Description

Selecting "Open SQL Script" or hitting F3 without any open script panes will open a new script pane instead of opening the "Choose SQL Script" window

### DBeaver Version

24.0.5

### Operating System

macOS 14

### Database and driver

MariaDB

### Steps to reproduce

1. Open DBeaver
2. Close all script panes
3. Press F3

A new script pane will open but the file select dialog will not

4. Press F3

Now the file select dialog opens. It should open even if there isn't an existing script pane as selecting a script to open creates a new pane anyway

**Comment by ShadelessFox** [comment]
Hello @JTMosaic,

I'm unable to reproduce the issue. What happens if you click on it explicitly?

![image](https://github.com/dbeaver/dbeaver/assets/35821147/719cbe7a-ba6d-4913-92c9-d47a6f5db06a)

**Comment by JTMosaic** [comment]
Very odd. 

I created a new Workspace folder and switched DBeaver to use it. I allowed it to create the sample SQLIte database. I connected to the sample database and hit F3 and it behaved as I described above: A new tab was created rather than the open file dialog. Pressing F3 again opens the file dialog.

I then switched back to my default workspace and now it is _no longer happening there._ This I cannot explain.

But is it still happening on the new workspace with only the SQLite sample database.

https://github.com/dbeaver/dbeaver/assets/9417720/3ffcce91-8653-4e4e-8273-bd1911e9a942

So, I guess I "fixed" it in my main workspace but I don't know what I did exactly.

**Comment by ShadelessFox** [comment]
Thanks for the provided information. We will try reproducing the issue on our side.

**Comment by E1izabeth** [comment]
I reproduced the case. However, I'm not sure it's an issue. This dialog appears, when you have more than one script for the selected database.

**Comment by JTMosaic** [comment]
I'm not sure I understand your comment.

The issue I raised was that, instead of opening the file dialog, a new query tab was opened.

**Comment by E1izabeth** [comment]
New script is opening, when you have the only script and there is nothing to choose.

**Comment by E1izabeth** [comment]
And on the video you provided, the only script is already opened. So, you get this dialog.

**Comment by JTMosaic** [comment]
I see what you are saying, Elizabeth.

If the only script you have for a server is "Script.sql" ***and*** that file is empty, then pressing F3 will open it automatically instead of presenting the "Choose SQL Script" dialog. 

This is confusing as it appears like F3 is just opening a new tab. It also ignores the fact that I may have scripts for other connections in the same project. 

If the file has any contents at all, even just a space, then the "Choose SQL Script" dialog will appear.

**Comment by ShadelessFox** [comment]
DBeaver automatically deletes empty scripts on close. It can be configured in Editors -> SQL Editor -> Scripts.

What happens in your video:
1. You ask DBeaver to open a script. DBeaver sees that no existing scripts exist for the selected connection, so it creates a new one without prompting you which script to choose.
2. When you close the editor without typing anything in it, DBeaver automatically deletes the file as described above
3. When you ask DBeaver to open a script again while having an editor open, DBeaver sees that you already have one script and prompts you to choose which file to open.

I don't think it's confusing if you know how it works.

**Comment by JTMosaic** [comment]
I don't really care if I have any saved scripts for that connection or not. I don't want to open a new script, I want to open a shared script. (i.e. the "Show scripts for all connections" checkbox is ignored)

I think it's confusing if you are using scripts across multiple connections in a project but don't happen to have any for a specific connection. 

Can the logic be updated to look for scripts in the **project** vs for the **connection**?

And there exists a separate function/key to create a new script if that is really what I wanted to do.

**Comment by ShadelessFox** [comment]
We don't want to change this behavior.

I have marked this discussion with the "wait for votes" label so people can vote for the proposal to be implemented.
